lib/internal/streams/readable.js

```diff
@@ -1083,9 +1090,12 @@ Readable.prototype.wrap = function(stream) {
   return this;
 };
 
-Readable.prototype[SymbolAsyncIterator] = function() {
+function asyncIteratorPrototype() {
   return streamToAsyncIterator(this);
-};
+}
+
+Readable.prototype[SymbolAsyncIterator] = asyncIteratorPrototype;
+Readable.prototype[originalAsyncIteratorPrototype] = asyncIteratorPrototype;
```

Comment:
Why is this needed at all? I'm a little bit lost.

Reply:
@mcollina spec compliance in case the prototype was tampered with

Reply:
I don't really understand what spec you are referring to, given this is in node streams not Web Stream.

Reply:
Ref to this https://github.com/nodejs/node/pull/48395#pullrequestreview-1470701774 and this https://github.com/nodejs/node/pull/48395#issuecomment-1586960623